plugins/CoreUpdater/CoreUpdater.php

```diff
@@ -73,6 +74,15 @@ public function dispatch()
             if (FrontController::shouldRethrowException()) {
                 throw new Exception("Piwik and/or some plugins have been upgraded to a new version. \n" .
                     "--> Please run the update process first. See documentation: http://piwik.org/docs/update/ \n");
+            } elseif ($module === 'API')  {
+
+                $outputFormat = strtolower(Common::getRequestVar('format', 'xml', 'string', $_GET + $_POST));
+                $response = new ResponseBuilder($outputFormat);
+                $e = new Exception('Database Upgrade Required. Your Piwik database is out-of-date, and must be upgraded before you can continue.');
+                echo $response->getResponseException($e);
+                Common::sendResponseCode(503);
```

Comment:
I will change it but seems to work anyway see test (maybe because of output buffering or so)